relay-server/src/services/buffer/envelope_buffer/mod.rs

```diff
@@ -131,6 +135,14 @@ impl PolymorphicEnvelopeBuffer {
             Self::InMemory(buffer) => buffer.has_capacity(),
         }
     }
+
+    /// Consumes the [`PolymorphicEnvelopeBuffer`] and shuts it down.
+    pub async fn shutdown(self) {
+        match self {
+            Self::Sqlite(buffer) => buffer.shutdown().await,
+            Self::InMemory(buffer) => buffer.shutdown().await,
```

Comment:
The in-memory impl should not drop any envelopes, so I would make this a noop and only implement shutdown for disk.